security/validateMessage_test.go

```diff
@@ -0,0 +1,736 @@
+		if err != nil {
+			t.Error(err)
+		}
+		rollCall, err := createEventRollCall(pubkey, privkey, rollCallCreation, start, rollCallName, lao_id)
+		if err != nil {
+			t.Error(err)
+		}
+		valid := RollCallOpenedIsValid(data, string(lao_id), rollCall)
+		if valid != true {
+			t.Errorf("Created rollcall Should be valid %#v", data)
+		}
+		err = CheckMessageIsValid(pubkey, privkey, data, witnessSignatures)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+//TestRollCallClosedIsValid checks that a message containing a createRollCall is valid at both message and data layer
+func TestRollCallClosedIsValid(t *testing.T) {
+	//increase nb of tests
+	for i := 0; i < 100; i++ {
+		pubkey, privkey := createKeyPair()
+		witnessSignatures := []message2.ItemWitnessSignatures{}
+		attendeesPks := [][]byte{}
+
+		rollCallCreation := time.Now().Unix()
+		rollCallName := "encore un roll call"
+		start := rollCallCreation + (MaxPropagationDelay / 4)
+		end := start + (MaxPropagationDelay / 4)
+		lao_id := []byte("12345")
+		data, err := createCloseRollCallNow(pubkey, privkey, rollCallCreation, start, end, rollCallName, lao_id, attendeesPks)
+		if err != nil {
+			t.Error(err)
+		}
+		rollCall, err := createEventRollCall(pubkey, privkey, rollCallCreation, start, rollCallName, lao_id)
+		if err != nil {
+			t.Error(err)
+		}
+		valid := RollCallClosedIsValid(data, string(lao_id), rollCall)
+		if valid != true {
+			t.Errorf("closed rollcall Should be valid %#v", data)
+		}
+		err = CheckMessageIsValid(pubkey, privkey, data, witnessSignatures)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+//TestRollCallCreatedIsValid checks that a message containing a createRollCall is valid at both message and data layer
+func TestRollCallCreatedIsValid(t *testing.T) {
+	//increase nb of tests
+	for i := 0; i < 100; i++ {
+		pubkey, privkey := createKeyPair()
+		witnessSignatures := []message2.ItemWitnessSignatures{}
+		var creation = time.Now().Unix()
+		start := creation + (MaxPropagationDelay / 2)
+		name := "RollCallNow"
+		lao_id := []byte("12345")
+		data, err := createRollCallNow(pubkey, privkey, creation, start, name, lao_id)
+		if err != nil {
+			t.Error(err)
+		}
+		valid := RollCallCreatedIsValid(data, string(lao_id))
+		if valid != true {
+			t.Errorf("Created rollcall Should be valid %#v", data)
+		}
+		err = CheckMessageIsValid(pubkey, privkey, data, witnessSignatures)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+//TestMeetingCreatedIsValid checks that a message containing a createMeeting is valid at both message and data layer
+func TestMeetingCreatedIsValid(t *testing.T) {
+	//increase nb of tests
+	for i := 0; i < 100; i++ {
+		pubkey, privkey := createKeyPair()
+		witnessSignatures := []message2.ItemWitnessSignatures{}
+		var creation = time.Now().Unix()
+		start := creation + (MaxPropagationDelay / 2)
+		end := creation + (MaxPropagationDelay / 2)
+		name := "someMeetintintinitin"
```

Comment:
mhhhh, don't you want to change that name ? That being said I just love it this way :)